be/src/olap/compaction.cpp

```diff
@@ -164,17 +173,11 @@ void Compaction::modify_rowsets() {
     _tablet->save_meta();
 }
 
-OLAPStatus Compaction::gc_unused_rowsets() {
+void Compaction::gc_output_rowset() {
```

Comment:
why not call gc_output_rowset in every succeed? It seems a new imported bug ?

Reply:
`gc_output_rowset()` should only be called when compaction succeed.

In #4212, we have implemented a new compaction logic, so that the input rowset will be deleted later. And we only need to delete the output rowset when failed.